lalrpop/src/lr1/codegen/test_all.rs

```diff
@@ -54,11 +55,23 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TestAll> {
                                         this.states,
                                         "super::super::super",
                                         this.out));
-            rust!(this.out,
-                  "pub use self::{}parse{}::parse_{};",
-                  this.prefix,
-                  this.start_symbol,
-                  this.user_start_symbol);
+            let pub_use = match this.grammar.nonterminals[&this.user_start_symbol].visibility {
```

Comment:
can we extract this into a helper function, like `emit_visibility` or something?